CMakeLists.txt

```diff
@@ -0,0 +1,234 @@
+    ${PROJECT_SOURCE_DIR}/pure-data/src/s_print.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/s_stuff.h
+    ${PROJECT_SOURCE_DIR}/pure-data/src/s_utf8.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/s_utf8.h
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_acoustics.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_arithmetic.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_array.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_connective.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_gui.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_interface.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_list.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_midi.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_misc.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_net.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_scalar.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_text.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_time.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_vexp.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_vexp.h
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_vexp_fun.c
+    ${PROJECT_SOURCE_DIR}/pure-data/src/x_vexp_if.c
+)
+
+#---------------------------------------#
+#        The Pure Data Extra Sources    #
+#---------------------------------------#
+set(PDEXTRASOURCES
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/bob~/bob~.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/bonk~/bonk~.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/choice/choice.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/fiddle~/fiddle~.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/loop~/loop~.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/lrshift~/lrshift~.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/pique/pique.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/sigmund~/sigmund~.c
+    ${PROJECT_SOURCE_DIR}/pure-data/extra/stdout/stdout.c
+)
+
+#---------------------------------------#
+#         The libpd Sources             #
+#---------------------------------------#
+
+set(LIBPDSOURCES
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/z_libpd.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/z_libpd.h
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/s_libpdmidi.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/x_libpdreceive.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/x_libpdreceive.h
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/z_hooks.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/z_hooks.h
+)
+set(LIBPDUTILSSOURCES
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/util/ringbuffer.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/util/ringbuffer.h
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/util/z_print_util.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/util/z_print_util.h
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/util/z_queued.c
+    ${PROJECT_SOURCE_DIR}/libpd_wrapper/util/z_queued.h
+)
+
+source_group(pd FILES ${PDSOURCES})
+source_group(pdextra FILES ${PDEXTRASOURCES})
+source_group(libpd FILES ${LIBPDSOURCES})
+source_group(libpdutils FILES ${LIBPDUTILSSOURCES})
+
+include_directories(${PROJECT_SOURCE_DIR}/libpd_wrapper)
+include_directories(${PROJECT_SOURCE_DIR}/pure-data/src)
+
+if(WIN32)
```

Comment:
This path is not generic (it was where I compiled the lib in my old project)

Reply:
Sure, I can remove it. I only kept it to keep it working for you.
What I do is to provide CMAKE_THREAD_LIBS_INIT and PTHREADS_INCLUDE_DIR as parameters to cmake, which allows me to link to a custom-built pthreads dll. This is the standard way to do it with the cmake pthreads module. I can document this on a "how to build with cmake".

Reply:
I think you should also use `set(CMAKE_THREAD_PREFER_PTHREAD True)`

Reply:
Yes, I also provide that flag to cmake